src/compiler/gcc.rs

```diff
@@ -479,6 +485,10 @@ where
         dependency_args.push(dep_flag);
         dependency_args.push(dep_target.unwrap_or_else(|| output.clone().into_os_string()));
     }
+    if let Some(true) = need_explicit_dep_argument_path {
+        dependency_args.push(OsString::from("-MF"));
+        dependency_args.push(Path::new(&output).with_extension("d").into_os_string());
+    }
```

Comment:
Does this behave properly in conjunction with `-E`?
I'm guessing it does because we don't cache `-E` due to it being a `TooHardFlag`, eh?

Reply:
looks like TooHardFlag will cause the sccache client to run the command directly, so it doesn't matter.